cloudflare/data_source_load_balancer_pools.go

```diff
@@ -0,0 +1,141 @@
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"origins": {
+							Type:     schema.TypeSet,
+							Computed: true,
+							Elem:     originsElem,
+						},
+
+						"enabled": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+
+						"minimum_origins": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+
+						"latitude": {
+							Type:     schema.TypeFloat,
+							Computed: true,
+						},
+
+						"longitude": {
+							Type:     schema.TypeFloat,
+							Computed: true,
+						},
+
+						"check_regions": {
+							Type:     schema.TypeSet,
+							Computed: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+
+						"description": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"monitor": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"notification_email": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"load_shedding": {
+							Type:     schema.TypeSet,
+							Computed: true,
+							Elem:     loadShedElem,
+						},
+
+						"created_on": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"modified_on": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceCloudflareLoadBalancerPoolsRead(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[DEBUG] Reading Load Balancer Pools")
+	client := meta.(*cloudflare.API)
+
+	poolsObj, err := client.ListLoadBalancerPools(context.Background())
+	if err != nil {
+		return fmt.Errorf("error listing load balancer pools: %w", err)
+	}
+
+	poolIds := make([]string, 0)
+	pools := make([]map[string]interface{}, 0, len(poolsObj))
+	for _, p := range poolsObj {
+		pools = append(pools, map[string]interface{}{
+			"id":                 p.ID,
+			"name":               p.Name,
+			"origins":            flattenLoadBalancerOrigins(d, p.Origins),
+			"enabled":            p.Enabled,
+			"minimum_origins":    p.MinimumOrigins,
+			"latitude":           p.Latitude,
+			"longitude":          p.Longitude,
+			"check_regions":      p.CheckRegions,
+			"description":        p.Description,
+			"monitor":            p.Monitor,
+			"notification_email": p.NotificationEmail,
+			"load_shedding":      flattenLoadBalancerLoadShedding(p.LoadShedding),
+			"created_on":         p.CreatedOn.Format(time.RFC3339Nano),
+			"modified_on":        p.ModifiedOn.Format(time.RFC3339Nano),
+		})
+		poolIds = append(poolIds, p.ID)
+	}
+
+	if err := d.Set("pools", pools); err != nil {
+		return fmt.Errorf("error setting load balaner pools: %w", err)
+	}
+
+	d.SetId(stringListChecksum(poolIds))
+	return nil
+}
```

Comment:
Would be great to have filtering handled there too.
Even if not needed for the specific usecase here, I believe that it's kind of important to provide ways to filter the pools, either directly from the API request (if the API allows) or with basic pattern matching directly in the provider. Other data resources have similar approaches, waf groups as an example: 
- https://github.com/cloudflare/terraform-provider-cloudflare/blob/master/cloudflare/data_source_waf_groups.go#L29-L46
- https://github.com/cloudflare/terraform-provider-cloudflare/blob/master/cloudflare/data_source_waf_groups.go#L92-L96
- https://github.com/cloudflare/terraform-provider-cloudflare/blob/master/cloudflare/data_source_waf_groups.go#L122-L128

Reply:
I can add this if @jacobbednarz agrees here.